File: Model/controllers/producto.js
var conexion = require('./conexion');

exports.list = function (req, res) {
    conexion.query("SELECT * FROM producto", (error, rows) => {
        if (error) {
            console.log(error);

        } else {
            return res.json(rows);
        }
    });

}

exports.get = function (req, res) {
    var codigo = req.params.codigo;
    conexion.query("SELECT * FROM producto WHERE codigo = ?", [codigo], (error, rows) => {
        if (error) {
            console.log(error);
        } else {
            return res.json(rows);
        }
    });

}

exports.add = function (req, res) {
    var json = JSON.parse(JSON.stringify(req.body));

    conexion.query("INSERT INTO producto set ?", json, (error, rows) => {
        if (error) {
            console.log(error);
        } else {
            console.log("Producto insertado: ", json);
        }
    });
    res.end();
}

exports.update = function (req, res) {
    var json = JSON.parse(JSON.stringify(req.body));
    var datos = {
        codigo: json.codigo,
        nombre: json.nombre,
        categoria: json.categoria,
        especificaciones: json.especificaciones,
        precio: json.precio,
        activo: json.activo,
        imagen: json.imagen
    };

    conexion.query("UPDATE producto set ? WHERE codigo = ?", [datos, json.codigo], (error, rows) => {
        if (error) {
            console.log(error);
        } else {
            console.log("Producto actualizado: ", datos);
        }
    });
    res.end();
}

exports.delete = function (req, res) {
    var codigo = req.params.codigo;

    conexion.query("DELETE FROM producto WHERE codigo = ?", [codigo], (error, rows) => {
        if (error) {
            console.log(error);
        } else {
            console.log("Producto borrado:", codigo);
        }
    });
    res.end();
}

File: Controller/Modelo/Producto.cs
namespace Intermodular_MVC_VladimirIriarte.Modelo
{
    public class Producto
    {
        public int codigo { get; set; }
        public string nombre { get; set; }
        public string categoria { get; set; }
        public string especificaciones { get; set; }
        public decimal precio { get; set; }
        public bool activo { get; set; }
        public string imagen { get; set; }

        public Producto()
        {
        }

        public Producto(int codigo, string nombre, string categoria
             , string especificaciones, decimal precio, bool activo, string imagen)
        {
            this.codigo = codigo;
            this.nombre = nombre;
            this.categoria = categoria;
            this.especificaciones = especificaciones;
            this.precio = precio;
            this.activo = activo;
            this.imagen = imagen;
        }

        public Producto(string nombre, string categoria
            , string especificaciones, decimal precio, bool activo, string imagen)
        {
            this.nombre = nombre;
            this.categoria = categoria;
            this.especificaciones = especificaciones;
            this.precio = precio;
            this.activo = activo;
            this.imagen = imagen;
        }
    }
}